#! /bin/sh /usr/share/dpatch/dpatch-run
## 29_macmini_repeat.dpatch by Mario Limonciello <superm1@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Flag repeat events on macmini (LP: #179608).

@DPATCH@
diff -urNad lirc-0.8.3~pre1~/daemons/hw_hiddev.c lirc-0.8.3~pre1/daemons/hw_hiddev.c
--- lirc-0.8.3~pre1~/daemons/hw_hiddev.c	2007-10-13 10:41:28.000000000 -0500
+++ lirc-0.8.3~pre1/daemons/hw_hiddev.c	2008-03-24 00:29:22.000000000 -0500
@@ -543,6 +543,11 @@
 	pre_code = 0;
 	main_code = (ev[0].value << 24) + (ev[1].value << 16) +
 	            (ev[2].value <<  8) + (ev[3].value <<  0);
+	if (main_code == 0)
+	{
+		/* some variants seem to send 0 to indicate repeats */
+		main_code = old_main_code;
+	}
 	if (old_main_code == main_code)
 	{
 		repeat_flag = 1;
